src/langchain_google_firestore/document_loader.py

```diff
@@ -0,0 +1,217 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+import itertools
+from typing import TYPE_CHECKING, Any, Iterator, List, Optional
+
+import more_itertools
+from google.cloud import firestore  # type: ignore
+from google.cloud.firestore import DocumentReference  # type: ignore
+from langchain_community.document_loaders.base import BaseLoader
+from langchain_core.documents import Document
+
+from .utility.document_converter import DocumentConverter
+
+USER_AGENT_LOADER = "langchain-google-firestore-python:document_loader"
+USER_AGENT_SAVER = "langchain-google-firestore-python:document_saver"
```

Comment:
would it be relevant to add the version of the sdk to the user agent?

Reply:
That will be done by SenseAI team in a follow-up PR